### PR TITLE
adding support for yaml topologies.  replicate hipster_shop in yaml.

### PR DESCRIPTION
**Pull request by karticus**
This PR adds support for yaml topology files.  **_(not yet for inline topologies)_**

Doing this to add more flexibility.  Specifically the ability to reuse yaml blocks with `&` and `<<`.

This is part of a research spike to determe if the `generatorreceiver` can be a general purpose synthetic data generation tool.  However, this seems independently useful outside the spike.  Hence, this PR.

### CODE REVIEW

File: collector/generatorreceiver/internal/topology/file.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package topology
 
 type File struct {
-	Topology *Topology `json:"topology"`
+	Topology *Topology `json:"topology" yaml:"topology"`
```
Comment (kristinapathak) [inline collector/generatorreceiver/internal/topology/file.go:4]: Why are we adding tags when this is the default expected key for yaml.Unmarshal?

Comment (karticus) [reply collector/generatorreceiver/internal/topology/file.go:4]: Valid question.  I did it just for "completeness" and to have parity with the `json:` annotations.

**Review by smithclay — APPROVED** [review]

File: collector/generatorreceiver/helper.go [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
 
-	byteValue, _ := ioutil.ReadAll(jsonFile)
+	byteValue, _ := ioutil.ReadAll(topoFile)
+
+	lowerTopoPath := strings.ToLower(topoPath)
+	if strings.HasSuffix(lowerTopoPath, ".json") {
+		err = json.Unmarshal(byteValue, &topo)
+	} else if strings.HasSuffix(lowerTopoPath, ".yaml") {
+		err = yaml.Unmarshal(byteValue, &topo)
+	}
```
Comment (kristinapathak) [inline collector/generatorreceiver/helper.go:41]: What happens if we pass a file that is neither json nor yaml?

Should we support `.yml` too?

Comment (karticus) [reply collector/generatorreceiver/helper.go:41]: Thanks.  This is updated.

Comment (karticus) [reply collector/generatorreceiver/helper.go:41]: meaning: we support `.yml` and we error if the suffix is none of `.json`, `.yaml`, `.yml`

**Review by kristinapathak — APPROVED** [review]

**Pull request merged**